### DISCUSSION TITLE
HA Supervised: razionalizzazione  switch/sensors/ etc

### DISCUSSION THREAD

**Discussion by labattazza** [indomus/forum#1261] [HUB: Home Assistant]
Ciao a tutti,

seguo con interesse da un paio di anni il sito, e da poco il forum.
Sto cercando di domotizzare la mia casa, usando un server ricavato da un NUC.

Dopo varie vicissitudini con installazione "Core" (problemi di aggiornamenti, dipendenze, passaggi di versione, DeConz ballerino) ho deciso di provare l'installazione Supervised.

Per cui ora ho HA installato tramite Docker. Sono riuscito a integrare sensori/pulsanti/luci sia Zwave sia ZigBee.
Siccome vorrei rendere tutto più ordinato e porre le basi per espansioni future vorrei organizzare il tutto dividendo la configurazione nei vari file switch.yaml, sensor.yaml e così via.

In questo modo dovrebbe esser anche possibile integrare meglio alcune componenti (come i pulsanti Aeotec Quad, ad esempio) che non si riesce a far funzionare solo tramite interfaccia.

Il mio problema è questo: il pairing delle varie luci e interruttori Tradrfi, le automazioni e così via li ho eseguiti tramite interfaccia grafica. Dopo essermi chiesto dove venissero salvate queste informazioni, ho fatto un po' di ricerche e ho visto che sono in 
/config/.storage/
core.config
core.config_entries
core.device_registry
core.entity_registry

nella cartella sono presenti anche altri file che riguardano hacs, lovelace e così via.

Ho cercato parecchio in rete per capire come "estrapolare" i dati e inserirli nei vari file di testo, come consigliato un po' ovunque, ma proprio non ci riesco.

Vorrei fare un lavoro il più possibile pulito, anche perché sto cercando di integrare una presa Shelly Plug S via MQTT e - come immaginavo - inserendo il codice in configuration.yaml (giusto per prova) sballo la configurazione degli altri switch.

Avrei anche una domanda riguardo a cosa sia meglio per i dispositivi Zigbee, ma magari è meglio se apro un altro topic.

Spero di non aver fatto una domanda troppo banale, ma questo è uno di quei problemi magari con soluzione "ovvia" che però blocca.

Grazie a chiunque potrà dare una mano!

Il file dove sono racchiusi tutti gli elementi

**Comment by snakuzzo** [comment]
Innanzi tutto avresti potuto evitare gli sbattimenti, semplicemente installando Core su Docker.
Quanto ai tuoi dubbi, i file YAML sono utilizzati per le configurazioni di entità "fatte a mano".
Le entità create tramite integrazioni (e quindi "in automatico") non le puoi prendere e togliere da lì per metterle nei file YAML.
Quelle configurazioni restano e resteranno sotto `.storage` dove TU non dovresti mai metterci le mani.

**Reply by labattazza** [reply]
ciao, grazie mille del chiarimento: infatti non mi sono azzardato a toccare nulla sotto .storage.

Riguardo all'installazione. Core su Docker... me l'ero persa come modalità di installazione. A tuo parere converrebbe ripartire sbarazzandosi di Supervisor e compagnia? Nella modalità attuale non ho avuto problemi di aggiornamenti, cosa che non mi era capitato prima (certamente per la mia relativa inesperienza)

**Reply by snakuzzo** [reply]
La scelta deve essere tua. Se sei proprio a digiuno su Liinux, Docker, etc... resta su HassOS (o comunque HA Supervised)

**Reply by labattazza** [reply]
Guarda, sto riflettendo molto su questo, e mi sto rileggendo le vostre preziose guide.
Non sono un supersmanettone su Linux, ma da una decina di anni abbondanti lo uso, e amministro anche tramite terminale.

Sul NUC ho installato anche un server Apache e Nextcloud, che uso per avere più copie dei miei file, e poterci accedere dall'esterno.

Il motivo per cui avevo deciso di passare a installazine supervised è che più di una volta, probabilmente per colpa mia, quando ho fatto aggiornamenti di versione mi sono trovato nell'impossibilità di andare avanti, per via di incompatibilità di versioni pyton et similia.
E cosa non secondaria, DeConz una volta installato a livello di sistema operativo attiva anceh il software di gestione grafica, e per non so quale motivo ogni tanto lo ritrovavo "attivo", andando in conflitto con l'integrazione in HA. Purtroppo tutto ciò accadeva più di un anno fa e non ricordo i dettagli, magari sono io che sbagliavo la configurazione.

Ora sto meditando di lasciar pedere il "sarchiapone", cancellare i vari container e ripartire con Core su Docker (sto cercando di comprendere bene quanto possa esser vantaggioso), o riprovare con "Core" puro.

Ad esempio uno dei problemi che ho è ascoltare l'audio di Spotify dall'uscita del NUC (Uso un DAC usb che offre un audio molto più pulito rispetto all'uscita integrata sulla scheda madre): ho il sospetto che il problema sia proprio nella "Catena" HA Suprvised/Hardware.

Dopo il lungo spiegone.. si, credo che nelle prossime sere mi metterò e ripartirò da capo, con calma, soffermandomi bene sui vari step.
Tra i vari dubbi, uno nonbanale: se reinstallo tutto, le varie lampade/switch/sensori Zigbee e Zwave vanno resettati per permettere una nuova integrazione?

Grazie mille per la pazienza :-D

**Reply by snakuzzo** [reply]
Beh...allor ti consiglio sicuramente di passare a core su docker...dimenticati del venv, con docker risolvi a priori tutti i problemi di cui parlavi prima a valle degli aggiornamenti. 

Sull'audio di spotify non saprei dirti.

Per quanto riguarda invece i device zigbee e zwave va visto un attimo nel dettaglio...
Se i device sono di integrazioni di Core, allora ti ritrovi tutto mantenendo la directory di configurazione.
Se invece sono relativi ad addon, andrebbe visto con calma e adattando la configurazione dell'addon all'eventuale altro container che andrai ad utilizzare al posto dell'addon
